libs/remote_storage/src/local_fs.rs

```diff
@@ -343,18 +344,19 @@ impl RemoteStorage for LocalFs {
 
     async fn delete(&self, path: &RemotePath) -> anyhow::Result<()> {
         let file_path = path.with_base(&self.storage_root);
-        if !file_path.exists() {
+        match fs::remove_file(&file_path).await {
+            Ok(()) => Ok(()),
+            // The file doesn't exist. This shouldn't yield an error to mirror S3's behaviour.
             // See https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html
             // > If there isn't a null version, Amazon S3 does not remove any objects but will still respond that the command was successful.
-            return Ok(());
-        }
-
-        if !file_path.is_file() {
-            anyhow::bail!("{file_path:?} is not a file");
+            Err(e) if e.kind() == ErrorKind::NotFound => Ok(()),
+            Err(e) => {
+                if !file_path.is_file() {
+                    anyhow::bail!("{file_path:?} is not a file");
+                }
```

Comment:
I went through the `git blame` -- the is_file check has always been with us from the first commit https://github.com/neondatabase/neon/blob/1d5abf1253b6e770fdcf15c37d17687b16afaf3e/pageserver/src/relish_storage/local_fs.rs#L87-L89.

I think we could just remove it without any loss of functionality so suggesting this:

```suggestion
```

But fine to leave it. If you have gained some idea as to why it is needed, we could add a comment and keep it. If you haven't gotten a clue, we could add a comment about that and keep it :) Without any discussion it just seems like it should be here, but I cannot imagine why.

Reply:
I'll just remove it then I think. I'd thought that it was added explicitly but if it's been there since the origin of the file then I suppose no. I didn't find it explicitly being requested in the thread of #582.